src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQManagedConnection.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using RabbitMQ.Client;
+
+namespace Enable.Extensions.Queuing.RabbitMQ.Internal
+{
+    internal class RabbitMQManagedConnection
+    {
+        public RabbitMQManagedConnection(IConnection connection)
+        {
+            Connection = connection;
+        }
+
+        public IConnection Connection { get; }
+
+        public int ReferenceCount { get;  private set; }
+
+        public bool Disposed { get; private set; }
+
+        public void DecrementReferenceCount()
+        {
+            ReferenceCount--;
```

Comment:
Is this line thread-safe? Do we need to user `Interlocked` or a similar construct here?